app/models/user_access.rb

```diff
@@ -42,23 +61,27 @@ def accessible_facilities(action)
       .or(Facility.where(facility_group: accessible_facility_groups(action)))
   end
 
-  def can?(action, model, record = nil)
-    if record&.is_a? ActiveRecord::Relation
-      raise ArgumentError, "record should not be an ActiveRecord::Relation."
-    end
+  def can?(action, model, operation, record = nil)
+    raise ArgumentError, "record should not be an ActiveRecord::Relation." if record&.is_a? ActiveRecord::Relation
+    raise ArgumentError, "#{operation} is not a supported operation." unless operation.in? VALID_OPERATIONS
 
     case model
```

Comment:
We dispatch on model, with each model getting its own `can_act_on_<model>` method. Keeps it clean